e2e/playwright/payUserFlow.spec.ts

```diff
@@ -0,0 +1,276 @@
+      published_flows: [publishedFlow],
+    } = await prodGqlPublic(
+      `
+        query GetPublishedFlowByFlowId($id: uuid!) {
+          published_flows(
+            where: {flow_id: {_eq: $id}},
+            limit: 1,
+            order_by: {created_at: desc}
+          ) {
+            id
+            data
+            summary
+          }
+        }
+      `,
+      {
+        id: productionFlowId,
+      }
+    );
+
+    userId = await insertTestUser(USER_EMAIL);
+    teamId = await createTeam(TEAM_SLUG);
+    flowId = await createFlow({ teamId, flowSlug: FLOW_SLUG });
+    publishedFlowId = await insertPublishedFlow({
+      publishedFlow: {
+        ...publishedFlow,
+        publisher_id: userId,
+        flow_id: flowId,
+      },
+    });
+  });
+
+  test.afterEach(async () => {
+    const analyticsIds = await getAnalyticsByFlowId(flowId);
+    await deleteTestData({
+      teamId,
+      flowId,
+      publishedFlowId,
+      userId,
+      analyticsIds,
+    });
+  });
+
+  test("Should pay within GOV.UK and send the application", async ({
+    page,
+  }) => {
+    await navigateToPayComponent({ page, mode: "preview" });
+
+    await page.getByText("Pay using GOV.UK Pay").click();
+    await fillGovUkCardDetails(page)(SUCCESS_CARD_NUMBER);
+    await page.locator("#confirm").click();
+    await page.getByLabel("Email address").fill(TEST_EMAIL);
+    await page.getByTestId("continue-button").click();
+
+    const { payment_id: paymentRef } = await awaitForPaymentResponse(
+      page,
+      TEAM_SLUG
+    );
+
+    expect(paymentRef).toBeTruthy();
+    expect(await page.getByText(paymentRef!).textContent()).toBeTruthy();
+  });
+});
+
+function clickButtonWithTextAndWait(page: Page) {
+  return async (text: string, waitFor: number) => {
+    await page.getByText(text, { exact: true }).click();
+    await page.getByTestId("continue-button").click();
+    await page.waitForTimeout(waitFor);
+  };
+}
+
+function uploadFileToInput(page: Page) {
+  return async ({ file, selector }: { selector?: string; file: Buffer }) => {
+    await page.setInputFiles(selector || "input", {
+      name: "test.png",
+      mimeType: "image/png",
+      buffer: file,
+    });
+  };
+}
+
+async function navigateToPayComponent({
+  page,
+  mode,
+}: {
+  page: Page;
+  mode: "preview" | "unpublished";
+}) {
+  const dummyImage = readFileSync(path.join(__dirname, "./assets/test.png"));
+
+  // XXX: When there is a page transition the animations take few ms to finish, due to this, if there are similar action (e.g Click "No")
+  // in a row, the second click will run twice on the same page and fail the test.
+  // To avoid this, we wait for the animation to finish before clicking the next button.
+  const clickButtonWithTextAndContinue = (text: string) =>
+    clickButtonWithTextAndWait(page)(text, DEFAULT_WAIT_TIME);
+
+  await page.goto(
+    `${URL}/buckinghamshire/apply-for-a-lawful-development-certificate/${mode}?analytics=false`
+  );
```

Comment:
rather than hardcoding this domain, can we template with `TEAM_SLUG` & `FLOW_SLUG` defined above?